src/mono/mono/mini/tramp-amd64.c

```diff
@@ -1097,6 +1104,12 @@ mono_arch_get_interp_to_native_trampoline (MonoTrampInfo **info)
 	for (i = 0; i < FLOAT_PARAM_REGS; ++i)
 		amd64_sse_movsd_reg_membase (code, i, AMD64_R11, MONO_STRUCT_OFFSET (CallContext, fregs) + i * sizeof (double));
 
+	/* store the values of context registers onto the stack and set the context registers from CallContext */
+	for (i = 0; i < CTX_REGS; i++) {
```

Comment:
Should this be done on all platforms, or only on apple platforms ?

Reply:
Only on Apple platforms targeting amd64 and arm64.